broker/nats/pubsub.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+// Package nats contains NATS message publisher implementation.
+package nats
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/mainflux/mainflux/broker"
+	"github.com/mainflux/mainflux/errors"
+	"github.com/nats-io/nats.go"
+)
+
+const prefix = "channel"
+
+var errNatsConn = errors.New("Failed to connect to NATS")
+
+var _ broker.Nats = (*pubsub)(nil)
+
+type pubsub struct {
+	conn *nats.Conn
+}
+
+// New returns NATS message publisher.
+func New(url string) (broker.Nats, error) {
```

Comment:
Same as here: https://github.com/mainflux/mainflux/pull/1080/files#r400795157, please use `Messenger` or `PubSub` interface. I prefer having 3 interfaces: `Publisher`, `Subscriber`, and `PubSub`. The broker would implement `PubSub`.